File: Card_Game_Engine/ClientApp/src/app/models/classes/create-game.ts
import {GameRule} from "./game-rule";
import {Card} from "./card";
import {create_game} from "../../create-game/create-game.namespace";

export class CreateGame {
  rules: GameRule[]
  startingDeck: Card[]
  startingPosition: number
  grid: Record<number, create_game.GridItem>
  width: number
  height: number

  constructor(
    rules: GameRule[],
    width: number, height: number,
    startingDeck: Card[] = [],
    grid: Record<number, create_game.GridItem> = {},
    startingPosition: number = 0
  ) {
    this.rules = rules;
    this.width = width;
    this.height = height;
    this.startingDeck = startingDeck;
    this.grid = grid;
    this.startingPosition = startingPosition;
  }
}

File: Card_Game_Engine/Models/Global/Classes/CreateGame.cs
using Card_Game_Engine.Models.Global.Classes;
using CreateNamespace = Card_Game_Engine.Models.CreateGame.Classes;

public class CreateGame
{
    public List<Rule> Rules { get; set; } = new();
    public List<CreateNamespace.Card> StartingDeck { get; set; } = new();
    public int StartingPosition { get; set; } = 0;
    public Dictionary<int, CreateNamespace.GridItem> Grid { get; set; } = new();
    public int Width { get; set; }
    public int Height { get; set; }
    public List<CreateNamespace.ManualTrigger> ManualTriggers { get; set; } = new();
    public List<Player> Players { get; set; } = new();
}